tests/test_hybrids.py

```diff
@@ -0,0 +1,865 @@
+    assert set(h.hybrid_meta['sc_point_gid']) == overlap
+
+
+def test_limits_and_ratios_output_values_with_fixed_ratio_col():
+    """Test that limits and ratios are properly applied in succession. """
+
+    limits = {'solar_capacity': 50, 'wind_capacity': 0.5}
+    ratio_cols = RatioColumns(
+        'solar_capacity', 'wind_capacity', fixed='wind_capacity'
+    )
+    ratios = {ratio_cols: (0.3, 3.6)}
+    bounds = (0.3 - 1e6, 3.6 + 1e6)
+
+    h = Hybridization(
+        SOLAR_FPATH, WIND_FPATH,
+        limits=limits,
+        ratios=ratios
+    ).run()
+
+    ratios = (h.hybrid_meta['hybrid_{}'.format(ratio_cols.num)]
+              / h.hybrid_meta['hybrid_{}'.format(ratio_cols.denom)])
+    assert np.all(ratios <= bounds[-1])
+    assert np.all(h.hybrid_meta['hybrid_{}'.format(ratio_cols.num)]
+                  <= h.hybrid_meta[ratio_cols.num])
+    assert np.allclose(h.hybrid_meta['hybrid_{}'.format(ratio_cols.fixed)],
+                       h.hybrid_meta[ratio_cols.fixed])
+    assert np.all(h.hybrid_meta['solar_capacity'] <= limits['solar_capacity'])
+    assert np.all(h.hybrid_meta['wind_capacity'] <= limits['wind_capacity'])
+
+
+def test_limits_and_ratios_output_values():
+    """Test that limits and ratios are properly applied in succession. """
+
+    limits = {'solar_capacity': 50, 'wind_capacity': 0.5}
+    ratio_cols = RatioColumns('solar_capacity', 'wind_capacity')
+    ratios = {ratio_cols: (0.3, 3.6)}
+    bounds = (0.3 - 1e6, 3.6 + 1e6)
+
+    h = Hybridization(
+        SOLAR_FPATH, WIND_FPATH,
+        limits=limits,
+        ratios=ratios
+    ).run()
+
+    ratios = (h.hybrid_meta['hybrid_{}'.format(ratio_cols.num)]
+              / h.hybrid_meta['hybrid_{}'.format(ratio_cols.denom)])
+    assert np.all(ratios.between(*bounds))
+    assert np.all(h.hybrid_meta['hybrid_{}'.format(ratio_cols.num)]
+                  <= h.hybrid_meta[ratio_cols.num])
+    assert np.all(h.hybrid_meta['hybrid_{}'.format(ratio_cols.denom)]
+                  <= h.hybrid_meta[ratio_cols.denom])
+    assert np.all(h.hybrid_meta['solar_capacity'] <= limits['solar_capacity'])
+    assert np.all(h.hybrid_meta['wind_capacity'] <= limits['wind_capacity'])
+
+
+@pytest.mark.parametrize("ratio_cols", [
+    ('solar_capacity', 'wind_capacity'),
+    ('solar_area_sq_km', 'wind_area_sq_km'),
+    ('solar_capacity', 'wind_capacity', 'solar_capacity'),
+    ('solar_capacity', 'wind_capacity', 'wind_capacity'),
+])
+@pytest.mark.parametrize("ratio, bounds", [
+    (0.5, (0.5 - 1e6, 0.5 + 1e6)),
+    (1, (1 - 1e6, 1 + 1e6)),
+    ((0.5, 1.5), (0.5 - 1e6, 1.5 + 1e6)),
+    ((0.3, 3.6), (0.3 - 1e6, 3.6 + 1e6))
+])
+def test_ratios_input(ratio_cols, ratio, bounds):
+    """Test that the hybrid meta limits the ratio columns correctly. """
+    ratio_cols = RatioColumns(*ratio_cols)
+    if ratio_cols.fixed is None:
+        h = Hybridization(
+            SOLAR_FPATH, WIND_FPATH,
+            ratios={ratio_cols: ratio}
+        ).run()
```

Comment:
Since you're already running the profile hybridization, maybe check to make sure that the profiles never exceed the ratio limits too? Or just check to make sure the profiles never exceed the stated hybrid capacity in the meta data.

Reply:
Good idea! Calculating the ratios of the profiles themselves is a bit tricky because there are 0 values that would have to be dealt with (not impossible but annoying). I decided to take the latter suggestion instead.